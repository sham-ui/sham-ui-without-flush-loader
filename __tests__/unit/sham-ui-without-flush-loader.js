import renderer, { compile } from 'sham-ui-test-helpers';
import WithoutFlushLoader from '../../src/without-flush-loader.sfc';

jest.useFakeTimers();

it( 'renders correctly', () => {
    const meta = renderer( WithoutFlushLoader, {
        loadingComponent: compile`Loading...`
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
    expect( meta.component.container.textContent ).toBe( '' );

    jest.runAllTimers();

    expect( meta.toJSON() ).toMatchSnapshot();
    expect( meta.component.container.textContent ).toBe( 'Loading...' );
} );

it( 'example from README', () => {
    const meta = renderer(
        compile( {
            WithoutFlushLoader
        } )`
            {% if loaded %}
                Content loaded!
            {% else %}
                <WithoutFlushLoader 
                    loadingComponent={{ComponentForLoading}}/>
            {% endif %}
        `
        ,
        {
            loaded: false,
            ComponentForLoading: compile`Loading...`
        }
    );
    expect( meta.toJSON() ).toMatchSnapshot();
    expect( meta.component.container.textContent ).toBe( '' );
    meta.component.update( {
        loaded: true
    } );
    expect( meta.component.container.textContent ).toBe( ' Content loaded! ' );
    meta.component.update( {
        loaded: false
    } );
    expect( meta.component.container.textContent ).toBe( '' );

    jest.runAllTimers();
    expect( meta.toJSON() ).toMatchSnapshot();
    expect( meta.component.container.textContent ).toBe( 'Loading...' );
} );
