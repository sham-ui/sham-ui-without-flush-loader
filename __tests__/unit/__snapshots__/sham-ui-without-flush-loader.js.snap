// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example from README 1`] = `
Object {
  "Options": Object {
    "ComponentForLoading": [Function],
    "ID": "component",
    "loaded": false,
  },
  "html": "
<!--0-->
<!--0-->
<!--0-->
",
}
`;

exports[`example from README 2`] = `
Object {
  "Options": Object {
    "ComponentForLoading": [Function],
    "ID": "component",
    "loaded": false,
  },
  "html": "
Loading...
<!--0-->
<!--0-->
<!--0-->
<!--0-->
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "delay": 100,
    "loadingComponent": [Function],
    "showLoader": false,
  },
  "html": "<!--0-->",
}
`;

exports[`renders correctly 2`] = `
Object {
  "Options": Object {
    "ID": "component",
    "delay": 100,
    "loadingComponent": [Function],
    "showLoader": true,
  },
  "html": "
Loading...
<!--0-->
<!--0-->
",
}
`;
